tests/test_mocks.py

```
@@ -0,0 +1,117 @@
+from translate_srt import translate_srt, TranslationDisplay, validate_srt_format
+from rich.console import Console
+from rich.live import Live
+
+@pytest.mark.asyncio
+async def test_openai_client_mocked():
+    """Test translation with mocked OpenAI client"""
+    mock_response = Mock()
+    mock_response.choices = [Mock(message=Mock(content="Translated text"))]
+    
+    with patch('translate_srt.read_srt') as mock_read:
+        with patch('translate_srt.client.chat.completions.create', return_value=mock_response) as mock_create:
```

Ruff (0.8.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)